src/sentry/utils/snowflake.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import OrderedDict
+
+from django.conf import settings
+
+
+@dataclass(frozen=True, eq=True)
+class SnowflakeBitSegment:
+    length: int
+    name: str
+
+    def validate(self, value):
+        if self.length <= 0:
+            raise Exception("The length should be a positive number")
+        if value >> self.length != 0:
+            raise Exception(f"{self.name} exceed max bit value of {self.length}")
+        return True
+
+
+SENTRY_EPOCH_START = datetime(2022, 4, 26, 0, 0).timestamp()
+ID_VALIDATOR = SnowflakeBitSegment(settings.SNOWFLAKE_ID_LENGTH, "Snowflake ID")
+
+VERSION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_VERSION_ID_LENGTH, "Version ID")
+TIME_DIFFERENCE = SnowflakeBitSegment(settings.SNOWFLAKE_TIME_DIFFERENCE_LENGTH, "Time difference")
+REGION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_ID_LENGTH, "Region ID")
+REGION_SEQUENCE = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_SEQUENCE_LENGTH, "Region sequence")
+
+
+def MSB_ordering(value):
+    lsb_ordering = f"{value:05b}"
```

Comment:
```suggestion
def MSB_ordering(value, width):
    lsb_ordering = f"{value:0{width}b}"
```

You'll need to account for the width, which may change.

Then use it as `MSB_ordering(1, settings.SNOWFLAKE_VERSION_ID_LENGTH)` for the default value.